File: Javascript/섹션2. (진행중) 1-2차원 배열 탐색/7. 봉우리 (중상)/AA.js
function solution(num, arr){

    let cnt=0
    for(let i=0;i<n;i++){
        for(let j=0;j<n;j++){

            let left=false
            let up=false
            let right=false
            let down=false

            if(i==0){ left=true }
            if(j==0){ up=true }
            if(i==n-1){ right=true }
            if(j==n-1) { down=true }

            mine=arr[i][j]
            if(!left){
                if(arr[i-1][j]<mine){
                    left=true
                }
                else{
                    left=false
                }
            }
            if(!up){
                if(arr[i][j-1]<mine){
                    up=true
                }
                else{
                    up=false
                }
            }
            if(!right){
                if(arr[i+1][j]<mine){
                    right=true
                }
                else{
                    right=false
                }
            }
            if(!down){
                if(arr[i][j+1]<mine){
                    down=true
                }
                else{
                    down=false
                }
            }
            if(left&&up&&right&&down){
                cnt++
            }
        }
    }
    return cnt
}
let arr=[]
let n=5
arr.push([5, 3, 7, 2, 3])
arr.push([3, 7, 1, 6, 1])
arr.push([7, 2, 5, 3, 4])
arr.push([4, 3, 6, 4, 1])
arr.push([8, 7, 3, 5, 2])
console.log(solution(n, arr))

File: Python/섹션 3 - (진행중) 탐색&시뮬레이션(string,1차원,2차원 리스트 탐색)/9. 봉우리 (중)/AA.py
import sys
# currentPath='C:\_Algorithm-Study-Folder\Algorithm-Study\Python\섹션 3 - (진행중) 탐색&시뮬레이션(string,1차원,2차원 리스트 탐색)\9. 봉우리'
# sys.stdin=open(currentPath+'\\input.txt', 'rt')

n=int(input())
arr=[list(map(int,input().split())) for _ in range(n)]
# arr=[list(map(int,input().split()))] * num

# print(arr)

cnt=0
for i in range(len(arr)):
    for j in range(len(arr)):
        left=999
        right=999

        up=999
        down=999

        if i==0:
            up=0
        else:
            up=arr[i-1][j]

        if j==0:
            left=0
        else:
            left=arr[i][j-1]

        if i==len(arr)-1:
            down=0
        else:
            down=arr[i+1][j]

        if j==len(arr)-1:
            right=0
        else:
            right=arr[i][j+1]

        if arr[i][j] > left:
            if arr[i][j] > right:
                if arr[i][j] > down:
                    if arr[i][j] > up:
                        cnt+=1
print(cnt)

        

